src/validation/rules/file_name.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright 2023 Helsing GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::*;
+use std::path::PathBuf;
+
+/// Ensure that file names match package names.
+///
+/// For example, a package named `physics` should be stored in `physics.proto`. A package named
+/// `physics.rotation` should be stored in `physics/rotation.proto`.
+#[derive(Debug, Clone, Default)]
+pub struct FileName;
+
+fn file_name(package_name: &str) -> PathBuf {
+    format!("{}.proto", package_name.replace('.', "/")).into()
+}
+
+impl Rule for FileName {
+    fn rule_info(&self) -> &'static str {
+        "making sure file names matches package names"
+    }
+
+    fn check_package(&mut self, package: &Package) -> Violations {
+        let candidate = file_name(&package.name);
+        let correct = candidate == package.file;
+        if correct {
+            Violations::default()
```

Comment:
```suggestion
            return Violations::default();
```

Reply:
i am aware 😅

Reply:
i proposed this, as i was proposing using _guards_ instead of "complex" if-else chains. there is no way in which you can remove the else branch entirely, without using the return keyword here.